rpc/send.go

```diff
@@ -208,7 +208,14 @@ func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.A
 // client is ready. On success, the reply is sent on the channel;
 // otherwise an error is sent.
 func sendOne(client *Client, timeout time.Duration, method string, args, reply interface{}, c chan interface{}) {
-	<-client.Ready
+	select {
+	case <-client.Ready:
+		break
+	case <-time.After(timeout):
+		c <- rpcError{fmt.Sprintf("rpc to %s timed out after %s as the client didn't become ready",
```

Comment:
I would shorten this a little,
`c <- rpcError{fmt.Sprintf("rpc to %s client not ready after %s`,`